packages/web-components/src/components/time-picker/time-picker.ts

```diff
@@ -0,0 +1,375 @@
+      _handleSlotChange: handleSlotChange,
+    } = this;
+
+    const labelClasses = classMap({
+      [`${prefix}--label`]: true,
+      [`${prefix}--visually-hidden`]: hideLabel,
+      [`${prefix}--label--disabled`]: disabled,
+    });
+
+    const timePickerClasses = classMap({
+      [`${prefix}--time-picker`]: true,
+      [`${prefix}--time-picker--invalid`]: invalid,
+      [`${prefix}--time-picker--warning`]: warning,
+      [`${prefix}--time-picker--readonly`]: readonly,
+      [`${prefix}--time-picker--${size}`]: size,
+      ...(className && { [className]: true }),
+    });
+
+    const inputClasses = classMap({
+      [`${prefix}--time-picker__input-field`]: true,
+      [`${prefix}--text-input`]: true,
+      [`${prefix}--time-picker__input-field-error`]: invalid || warning,
+      ...(className && { [className]: true }),
+    });
+    const label = labelText
+      ? html`<label class="${labelClasses}">${labelText}</label>`
+      : null;
+
+    return html`
+      <div class="${prefix}--form-item ">
+        ${label}
+        <div class="${timePickerClasses}">
+          <div class="${prefix}--time-picker__input">
+            <input
+              class="${inputClasses}"
+              ?data-invalid="${invalid}"
+              ?disabled="${disabled}"
+              maxlength="${ifNonEmpty(maxLength)}"
+              name="${ifNonEmpty(this.name)}"
+              pattern="${ifNonEmpty(pattern)}"
+              placeholder="${ifNonEmpty(placeholder)}"
+              ?readonly="${readonly}"
+              type="${ifNonEmpty(type)}"
+              .value="${value}"
+              @input="${handleInput}"
+              @click="${handleClick}"
+              @blur="${handleBlur}" />
+            ${invalid || warning
+              ? html`
+                  <div class="${prefix}--time-picker__error__icon">
+                    ${invalid
+                      ? WarningFilled16({
+                          class: `${prefix}--checkbox__invalid-icon`,
+                        })
+                      : WarningAltFilled16({
+                          class: `${prefix}--text-input__invalid-icon--warning`,
+                        })}
+                  </div>
+                `
+              : null}
+          </div>
+          <slot
+            name="time-picker-select"
+            @slotchange="${handleSlotChange}"></slot>
+        </div>
+        ${invalid || warning
+          ? html`
+              <div class="${prefix}--form-requirement">
+                ${invalid ? invalidText : warningText}
+              </div>
+            `
+          : null}
+      </div>
+    `;
+  }
+  updated(changedProperties) {
+    super.updated && super.updated(changedProperties);
+
+    // Propagate properties to children whenever readonly or disabled changes
+    if (
+      changedProperties.has('readonly') ||
+      changedProperties.has('disabled') ||
+      changedProperties.has('size')
+    ) {
+      this.updateChildProperties();
+    }
+  }
```

Comment:
I think we can get rid of the `updateChildProperties` and the `handleSlotChange` and utilize how checkbox-group is handling passing down the props: https://github.com/carbon-design-system/carbon/blob/main/packages/web-components/src/components/checkbox/checkbox-group.ts#L130-L138

and add like a 
```
  static get selectorTimePickerSelect() {
    return `${prefix}-time-picker-select`;
  }
```